pkgs/applications/misc/slstatus/default.nix

```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchgit, pkgconfig, writeText, libX11, conf ? null }:
+
+with stdenv.lib;
+
+stdenv.mkDerivation rec {
+  name = "slstatus-${version}";
+  version = "unstable-2018-03-21";
+
+  src = fetchgit {
+    url = https://git.suckless.org/slstatus;
+    rev = "e79d4932ea413c277425ff92c456741e0730c3d6";
+    sha256 = "1c1al3vkywv0rpg4qaa67wvnc1v5ggvlsc1ynq11cr8gs4msvcl0";
+  };
+
+  configFile = optionalString (conf!=null) (writeText "config.def.h" conf);
+  preBuild = optionalString (conf!=null) "cp ${configFile} config.def.h";
```

Comment:
I think it would be better to not do the String -> Path conversions here, because it could be the case that the user wants to provide their own file, so they'd have to pass `builtins.readFile ./my-config.def.h`, which would only get converted right back to a file here.

So I suggest making `conf` a path parameter, removing the configFile attribute and just using `cp ${conf} config.def.h`. Also an assertion that checks for `conf` actually being a path (starts with "/") would be nice, along with a simple message.

And probably `configFile` would be a better name then, there's a couple uses of a `configFile` argument throughout nixpkgs already.

Reply:
I have to admit that I'm not a big fan from this solution, which I adapted from the `nixpkgs/pkgs/applications/misc/st/default.nix`-file. Isn't it better if there is at least some regularity? That's the reason why I choose this way.